### ISSUE TITLE
"Your Steam account is already associated with another member."

### ISSUE THREAD

**Issue by ImStayFrosty** [Lavoaster/IP.Board-Steam-Authentication-Method#115]
People can no longer sign-in with steam after I updated to the application instead of the plugin. I followed all the steps and deleted the files it said to but nobody can use steam to login as it tries to make another account.

**Comment by itsAliAvali** [comment]
How recently did you upgrade?

**Comment by ImStayFrosty** [comment]
Within the last hour, I have run all background tasks as well.

**Comment by ImStayFrosty** [comment]
![image](https://user-images.githubusercontent.com/10207200/41430080-139617a8-6fdd-11e8-8835-b077e44d818e.png)

**Comment by ImStayFrosty** [comment]
INSERT INTO `core_login_links` ( `token_login_method`, `token_member`, `token_identifier`, `token_linked` ) VALUES ( 10, 2, '76561198122638261', 1 ), ( 10, 5, '76561198103239748', 1 ), ( 10, 6, '76561198197612587', 1 ), ( 10, 12, '76561198009784986', 1 ), ( 10, 13, '76561198065124435', 1 ), ( 10, 14, '76561198124059936', 1 ), ( 10, 15, '76561198240230526', 1 ), ( 10, 16, '76561198152183314', 1 ), ( 10, 17, '76561198257766619', 1 ), ( 10, 18, '76561198061457818', 1 ), ( 10, 19, '76561198351086435', 1 ), ( 10, 20, '76561198075782246', 1 ), ( 10, 21, '76561198084970276', 1 ), ( 10, 22, '76561198177113481', 1 ), ( 10, 23, '76561198096910438', 1 ), ( 10, 24, '76561198223321852', 1 ), ( 10, 25, '76561197978016370', 1 ), ( 10, 26, '76561198149824470', 1 ), ( 10, 27, '76561198082089914', 1 ), ( 10, 29, '76561198081843133', 1 ), ( 10, 30, '76561198065533803', 1 ), ( 10, 31, '76561198118907758', 1 ), ( 10, 32, '76561198078129677', 1 ), ( 10, 33, '76561198105346755', 1 ), ( 10, 34, '76561198114955662', 1 ), ( 10, 35, '76561198321276885', 1 ), ( 10, 36, '76561198044478761', 1 ), ( 10, 37, '76561198182494662', 1 ), ( 10, 38, '76561198101264026', 1 ), ( 10, 39, '76561198069510590', 1 ), ( 10, 40, '76561198208453412', 1 ), ( 10, 41, '76561198073164415', 1 ), ( 10, 42, '76561198033617523', 1 ), ( 10, 43, '76561198183228931', 1 ), ( 10, 44, '76561198060040901', 1 ), ( 10, 45, '76561198149839711', 1 ), ( 10, 46, '76561198280882566', 1 ), ( 10, 47, '76561198063604290', 1 ), ( 10, 48, '76561198043173794', 1 ), ( 10, 49, '76561198065473696', 1 ), ( 10, 50, '76561198122334082', 1 ), ( 10, 52, '76561198050124085', 1 ), ( 10, 53, '76561198031488159', 1 ), ( 10, 54, '76561198147112715', 1 ), ( 10, 55, '76561198117059598', 1 ), ( 10, 56, '76561198115211245', 1 ), ( 10, 57, '76561198207447961', 1 ), ( 10, 58, '76561198134677610', 1 ), ( 10, 59, '76561198347730430', 1 ), ( 10, 60, '76561198159450349', 1 ), ( 10, 61, '76561198145662177', 1 ), ( 10, 62, '76561198065057039', 1 ), ( 10, 63, '76561198124921345', 1 ), ( 10, 64, '76561198306872428', 1 ), ( 10, 65, '76561198190260459', 1 ), ( 10, 66, '76561198285474303', 1 ), ( 10, 67, '76561198011859093', 1 ), ( 10, 68, '76561198069748201', 1 ), ( 10, 69, '76561198286285671', 1 ), ( 10, 70, '76561198314897600', 1 ), ( 10, 71, '76561198177288286', 1 ), ( 10, 72, '76561198068049243', 1 ), ( 10, 73, '76561198107300329', 1 ), ( 10, 74, '76561198130671512', 1 ), ( 10, 75, '76561198023962184', 1 ), ( 10, 76, '76561198134708872', 1 ), ( 10, 77, '76561198122200594', 1 ), ( 10, 78, '76561198198530769', 1 ), ( 10, 79, '76561198058562944', 1 ), ( 10, 80, '76561198106054757', 1 ), ( 10, 81, '76561198188701723', 1 ), ( 10, 82, '76561198358292264', 1 ), ( 10, 83, '76561198185642305', 1 ), ( 10, 84, '76561197979652941', 1 ), ( 10, 85, '76561198097816871', 1 ), ( 10, 86, '76561198105163497', 1 ), ( 10, 87, '76561198095248164', 1 ), ( 10, 88, '76561198006377714', 1 ), ( 10, 89, '76561198040529843', 1 ), ( 10, 90, '76561198357810426', 1 ), ( 10, 91, '76561198100983448', 1 ), ( 10, 92, '76561198121440755', 1 ), ( 10, 93, '76561198125431595', 1 ), ( 10, 94, '76561198156634273', 1 ), ( 10, 95, '76561198119900268', 1 ), ( 10, 96, '76561198100699427', 1 ), ( 10, 97, '76561198053219309', 1 ), ( 10, 98, '76561198074997032', 1 ), ( 10, 99, '76561198091651775', 1 ), ( 10, 100, '76561198084717237', 1 ), ( 10, 101, '76561198359976692', 1 ), ( 10, 102, '76561198372092334', 1 ), ( 10, 103, '76561198144812537', 1 ), ( 10, 104, '76561198088962641', 1 ), ( 10, 105, '76561198110583228', 1 ), ( 10, 106, '76561198139510546', 1 ), ( 10, 107, '76561198133195115', 1 ), ( 10, 108, '76561198076149901', 1 ), ( 10, 110, '76561197994609880', 1 ), ( 10, 111, '76561198082953811', 1 )
Duplicate entry '10-2' for key 'PRIMARY'

**Comment by PRAWKC** [comment]
Getting the same issue as you, I changed the token_login_method because or else if someone was already connected with steam then when they log in it would create a new account that was bugged. I also have the exact same issue in my admincp "Migrating Steam User Login Data" Anyone find a fix for this?

Edit: Issue has been resolved

**Comment by ImStayFrosty** [comment]
Yeah, I never got a response either. Guess I have to disable steam login now.

**Comment by itsAliAvali** [comment]
with this being a dedicated application now instead of a plugin, have any of you guys removed the old plugin and accompanying files?

**Comment by itsAliAvali** [comment]
during migration it should have picked up on duplicates.... right? @Lavoaster ?

**Comment by ImStayFrosty** [comment]
Once the plugins list the steam with plugin has disappeared, I assumed it deleted it by itself. I cannot find any trace of the plugin anywhere. I even deleted the files was supposed to delete like it said in the readme.md

**Comment by ImStayFrosty** [comment]
* On the plugins list the steam auth plugin has disappeared

Apologies as I typed this msg on my iPhone in the middle of the night.

**Comment by ImStayFrosty** [comment]
Still no solution.

**Comment by LavaToaster** [comment]
@madman32395 Yes, it is supposed to be picked up now. I don't have the time to create and test query for this but it should be possible to run something like 

```
SELECT count(*) total, steamid FROM core_members GROUP BY steamid HAVING total > 1
```

and clear out any with duplicates.

```
DELETE FROM core_login_links WHERE method_id = <login_method_id>
```

Follow it up by clearing out the `core_login_links` (this might not be the right name) table where the login method id matches the steam one and then do something like 

```
INSERT INTO core_login_links ( token_login_method, token_member, token_identifier, token_linked) SELECT <login_method_id>, member_id, steamid, 1 FROM core_members WHERE steamid > 0
```



I don't know when I'll be able to get to writing and fully testing it these queries, but hopefully that should provide enough guidance for someone proficient with SQL to write one. Since the plugin works fine, just the migration of data seems to be a bit wonky for a few.

**Comment by ImStayFrosty** [comment]
Is there any way of completely removing everything created by steam auth. I would just rather restart than trying to migrate the data. Too many people have created duplicate accounts and such I would rather just let people re-link their accounts if possible @Lavoaster.

**Comment by peter-r-g** [comment]
Is there a fix for this? I just upgraded and now it's broken for me. It's quite important for the data to be migrated since almost everyone has signed up with steam

**Comment by narapon** [comment]
The manual SQL statements work

Before clearing out core_login_links, take note of the existing token_login_method number for steam

edit. backup before trying of course! 😃
